provider/cmd/serviceLogs.go

```diff
@@ -26,56 +25,44 @@ func serviceLogsCmd() *cobra.Command {
 		},
 	}
 
-	mcli.AddBidIDFlags(cmd.Flags())
-	mcli.MarkReqBidIDFlags(cmd)
-
-	cmd.Flags().String("service", "", "")
-	_ = cmd.MarkFlagRequired("service")
+	addServiceFlags(cmd)
 
 	cmd.Flags().BoolP("follow", "f", false, "Specify if the logs should be streamed. Defaults to false")
 	cmd.Flags().Int64P("tail", "t", -1, "The number of lines from the end of the logs to show. Defaults to -1")
 	cmd.Flags().String("format", "text", "Output format text|json. Defaults to text")
+
 	return cmd
 }
 
 func doServiceLogs(cmd *cobra.Command) error {
-	cctx := client.GetClientContextFromCmd(cmd)
-
-	// TODO - changeme not to use cmd.Flags, use viper
-	addr, err := mcli.ProviderFromFlagsWithoutCtx(cmd.Flags())
+	cctx, err := client.ReadTxCommandFlags(client.GetClientContextFromCmd(cmd), cmd.Flags())
 	if err != nil {
 		return err
 	}
 
-	svcName, err := cmd.Flags().GetString("service")
+	svcName, err := cmd.Flags().GetString(FlagService)
 	if err != nil {
 		return err
 	}
 
-	outputFormat, err := cmd.Flags().GetString("format")
+	prov, err := providerFromFlags(cmd.Flags())
 	if err != nil {
 		return err
 	}
 
-	if outputFormat != "text" && outputFormat != "json" {
-		return errors.Errorf("invalid output format %s", outputFormat)
-	}
-
-	pclient := pmodule.AppModuleBasic{}.GetQueryClient(cctx)
-	res, err := pclient.Provider(context.Background(), &ptypes.QueryProviderRequest{Owner: addr.String()})
+	dseq, gseq, oseq, err := parseLeaseFromFlags(cmd.Flags())
 	if err != nil {
 		return err
 	}
 
-	provider := &res.Provider
-	gclient := gateway.NewClient()
-
-	bid, err := mcli.BidIDFromFlagsWithoutCtx(cmd.Flags())
+	outputFormat, err := cmd.Flags().GetString("format")
 	if err != nil {
 		return err
 	}
 
-	lid := mtypes.MakeLeaseID(bid)
+	if outputFormat != "text" && outputFormat != "json" {
+		return errors.Errorf("invalid output format %s", outputFormat)
```

Comment:
It'd be helpful to indicate what the valid output formats are in this error message.